src/webauth/index.js

```diff
@@ -49,9 +51,11 @@ export default class WebAuth {
 
         const { clientId, client, agent } = this
         const {nonce, state, verifier} = await agent.generateRequestParams()
-
+        const { extraQueryParameters, ...remainingOptions } = options;
+    
         let requestParams = {
-            ...options,
+            ...remainingOptions,
+            ...extraQueryParameters,
```

Comment:
I would use:

`... (extraQueryParameters || {})`

to avoid errors if no extra params are defined

Reply:
Thanks, fixed